prettyprinter/src/Data/Text/Prettyprint/Doc/Internal.hs

```diff
@@ -1467,8 +1467,8 @@ removeTrailingWhitespace = go (RecordedWhitespace [] 0)
     commitSpaces [] 0 = id
     commitSpaces [] 1 = SChar ' '
     commitSpaces [] n = SText n (T.replicate n " ")
-    commitSpaces [i] _ = SLine i
-    commitSpaces (_:is) _ = SLine 0 . commitSpaces is 0
+    commitSpaces [i] n = SLine i . commitSpaces [] n
+    commitSpaces (_:is) n = SLine 0 . commitSpaces is n
```

Comment:
Good catch! I remember thinking »is 0 really the right value here«, but forgot about it when the tests passed. Woops!